languages/elixir/concepts/tail-call-recursion/about.md

```diff
@@ -1 +1,37 @@
-TODO: add information on tail-call-recursion concept
+A function is [tail-recursive][recursion-tc] if the _last_ thing executed by the function is a call to itself.
+
+Each time any function is called, a _stack frame_ with its local variables, arguments etc. is put on top of [the function call stack][call-stack]. When a function returns, the stack frame is removed from the stack.
+
+Tail-recursive functions allow for [tail call optimization][tail-call-optimization]. It's an optimization that allows reusing the last stack frame by the next function call because the previous function is guaranteed not to need it anymore. This mitigates concerns of overflowing the function call stack, which is a situation when there are so many frames on the function call stack that there isn't any memory left to create another one.
```

Comment:
My CS knowledge is a bit rusty... I hope what I added here is accurate.

Reply:
It looks good to me. :+1: